lib/datasource/datasource.ts

```diff
@@ -0,0 +1,55 @@
+import { ExternalHostError } from '../types/errors/external-host-error';
+import { Http, HttpError } from '../util/http';
+import {
+  DatasourceApi,
+  DigestConfig,
+  GetReleasesConfig,
+  ReleaseResult,
+} from './types';
+
+export abstract class Datasource implements DatasourceApi {
+  constructor() {
+    this.http = new Http(this.getId());
+  }
+
+  abstract id: string;
+
+  caching: boolean;
+
+  defaultConfig: Record<string, unknown>;
+
+  customRegistrySupport: boolean;
+
+  defaultRegistryUrls: string[];
+
+  defaultVersioning: string;
+
+  registryStrategy: 'first' | 'hunt' | 'merge';
+
+  getId(): string {
+    return this.id;
+  }
```

Comment:
Do we need this wrapper function? i think we can directly use `this.id`

Reply:
It's not possible to use properties directly in the constructor

Reply:
Then i think it's better to pass id via constructor and making it readonly in datasource base class